common/djangoapps/course_category/tasks.py

```diff
@@ -27,3 +29,53 @@ def task_add_categories(category_names, course_id):
         category = CourseCategory.objects.filter(name=category_name).first()
         if category:
             category.courses.add(CourseKey.from_string(course_id))
+
+
+@task
+def task_enroll_category(user_id, slug, is_secure):
+    """
+    Task for enrolling to category courses after pressing "Bulk Enroll" button.
+
+    Args:
+        user_id (int) - current user;
+        slug (str) - category to enroll slug;
+        is_secure (True/False) - request.is_secure()
+    """
+    from django.contrib.auth.models import User
+
+    from course_category.utils import enroll_category, send_email_with_enroll_status
+    from .models import CourseCategory
+
+    user = User.objects.filter(id=user_id).first()
+    category = CourseCategory.objects.filter(slug=slug).first()
+
+    enroll_result = enroll_category(user, category)
+
+    enrolled_courses_names = [course.display_name for course in enroll_result['enrolled']]
```

Comment:
Why do you need to create separate lists?

Reply:
Replaced with dict.